chore: Add missing javadocs in `TokenService`

Neeharika-Sompalli opened the pull request:
Fixes #13048 

Add missing javadocs in `TokenService`

iwsimon commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java:
```diff
@@ -146,6 +147,7 @@ public Account get(@NonNull final AccountID accountID) {
      * If no such account exists, returns {@code Optional.empty()}
      *
      * @param id - the number of the account to be retrieved.
+     * @return the account with the given account number, or null if no such account exists.
```
Maybe add some words so the comment is not the same as get(AccountID)?

kimbor reviewed (commented):
I'm not sure how nit-picky we want to be with these. Here are some things that could be improved, but feel free to ignore if this is more work than is deemed worthwhile.

- Need `package-info.java` files for all packages. Currently only `com.hedera.node.app.service.token.impl` has one.
- `BlocklistParser.BlockedInfo` needs a class description
- `TransferContext.validateHbarAllowances` has a "(FUTURE) REMOVE THIS" note. This note should be put in regular comments with a @future tag, instead of in javadocs
- We are inconsistent in punctuation. All class descriptions should end with a period. ApproveAllowanceValidator, BlockListParser, some others currently do not.
- All method descriptions should end in a period. Currently there are lots that do not.
- CryptoCreateValidator class description should read “Tokensymbol, etc. It is used”
- CryptoCreateValidator.tooManyAutoAssociations description should be capitalized and end in a period.
- Only class descriptions up to the first period are included in the class summary. Therefore the summary for TokenHandlerHelper looks like “Class for retrieving objects in a certain context, e.g.”. It would be better to either escape the periods in e.g. or just write “for example” instead.

iwsimon commented on hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateNftsSuite.java:
```diff
@@ -143,14 +142,13 @@ final HapiSpec updateMetadataOfNfts() {
                 .then(
                         getTokenNftInfo(NON_FUNGIBLE_TOKEN, 7L)
                                 .hasSerialNum(7L)
+                                .logged()
```
debug code?

Neeharika-Sompalli replied:
Removed. Thanks

iwsimon reviewed (dismissed):
LGTM. Couple comments. Thanks @Neeharika-Sompalli

Neeharika-Sompalli commented:
> I'm not sure how nit-picky we want to be with these. Here are some things that could be improved, but feel free to ignore if this is more work than is deemed worthwhile.
> 
> * Need `package-info.java` files for all packages. Currently only `com.hedera.node.app.service.token.impl` has one.
> * `BlocklistParser.BlockedInfo` needs a class description
> * `TransferContext.validateHbarAllowances` has a "(FUTURE) REMOVE THIS" note. This note should be put in regular comments with a @future tag, instead of in javadocs
> * We are inconsistent in punctuation. All class descriptions should end with a period. ApproveAllowanceValidator, BlockListParser, some others currently do not.
> * All method descriptions should end in a period. Currently there are lots that do not.
> * CryptoCreateValidator class description should read “Tokensymbol, etc. It is used”
> * CryptoCreateValidator.tooManyAutoAssociations description should be capitalized and end in a period.
> * Only class descriptions up to the first period are included in the class summary. Therefore the summary for TokenHandlerHelper looks like “Class for retrieving objects in a certain context, e.g.”. It would be better to either escape the periods in e.g. or just write “for example” instead.

@kimbor Thank you.

> Need package-info.java files for all packages. Currently only com.hedera.node.app.service.token.impl has one.

Not sure what is the decided convention for this

Addressed all comments except ending witg period. Will fix those as I go through the files again.

mhess-swl commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java:
```diff
@@ -221,6 +250,13 @@ private void validateAllowanceCount(
     }
 
+    /**
+     * Validates the token basic fields like token type, token relation exists.
```
```suggestion
     * Validates that token basic fields like token type or token relation exist.
```

Neeharika-Sompalli replied:
Fixed.

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableTokenRelationStore.java:
```diff
@@ -99,6 +99,7 @@ public void remove(@NonNull final TokenRelation tokenRelation) {
      *
      * @param accountId - the number of the account to be retrieved
      * @param tokenId   - the number of the token to be retrieved
+     * @return the token relation with the given token number and account number, or {@code Optional.empty()} if no such
```
Added `token relation exists`

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AdjustFungibleTokenChangesStep.java:
```diff
@@ -54,6 +54,11 @@ public class AdjustFungibleTokenChangesStep extends BaseTokenHandler implements
 
+    /**
+     * Constructs the step with the CryptoTransferTransactionBody and the topLevelPayer.
+     * @param op - the CryptoTransferTransactionBody
+     * @param topLevelPayer - the payer account
```
removed

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java:
```diff
@@ -57,6 +57,10 @@
 
+    /**
+     * Constructs the step with the operation.
+     * @param op - the operation
```
Removed

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/fixtures/FakeNodeStakeUpdateRecordBuilder.java:
```diff
@@ -21,8 +21,15 @@
 
+/**
+ * Fake Node Stake Update Record Builder
+ *
```
Removed

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/staking/EndOfStakingPeriodUpdaterTest.java:
```diff
@@ -350,16 +353,32 @@ void calculatesMidnightTimeCorrectly() {
+     */
     public static final EntityNumber NODE_NUM_8 =
             EntityNumber.newBuilder().number(8).build();
+    /** Staking info for node 1. */
```
Ah yeah 🤷

Neeharika-Sompalli commented on hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/Units.java:
```diff
@@ -16,6 +16,9 @@
 package com.hedera.node.app.service.token;
 
+/**
+ * Utility class for units
```
This will be utility class for all units imo.

mhess-swl replied:
Ah got it. That may be a good thing to clarify in this javadoc

Neeharika-Sompalli commented on hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/api/StakingRewardsApi.java:
```diff
@@ -37,10 +37,25 @@
+     */
     Logger log = LogManager.getLogger(StakingRewardsApi.class);
+    /**
+     * Constants for time conversion from minutes to seconds
+     */
     int MINUTES_TO_SECONDS = 60;
+    /**
+     * Constants for time conversion from minutes to milliseconds
+     */
     long MINUTES_TO_MILLISECONDS = 60_000L;
+    /**
+     * Constants for daily staking period in minutes
+     */
     long DAILY_STAKING_PERIOD_MINS = 1440L;
+    /**
+     * Constants for UTC time zone
+     */
```
Yes changed. Thanks

Neeharika-Sompalli commented on hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateNftsSuite.java:
```diff
@@ -143,14 +142,13 @@ final HapiSpec updateMetadataOfNfts() {
                                 .hasSerialNum(7L)
+                                .logged()
                                 .hasMetadata(ByteString.copyFromUtf8("g")),
-                        tokenUpdateNfts(NON_FUNGIBLE_TOKEN, NFT_TEST_METADATA, List.of(7L))
-                                .signedBy(GENESIS)
-                                .hasKnownStatus(INVALID_SIGNATURE),
```
Oh sorry, was testing something else and removed. Great catch. Thanks @mhess-swl

Neeharika-Sompalli replied:
I think it doesn't show as a diff now. Because its same as develop 😄

Neeharika-Sompalli commented on hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:
```diff
@@ -79,6 +79,10 @@
 
+    /**
+     * Default constructor for injection.
+     * @param validator - the token supply change ops validator
```
Removed

iwsimon reviewed (approved):
LGTM. Thanks @Neeharika-Sompalli

mhess-swl commented:
There's a lot of less-than-exciting work here; thanks for taking care of it @Neeharika-Sompalli ! 🙌
